### ISSUE TITLE
Question: trouble with decoding LibriSpeech decoding demo

### ISSUE THREAD

**Issue by vergilus** [tensorflow/tensor2tensor#841]
### Description
I trained Librispeech problem days ago and follow the notebook demo in  [https://github.com/tensorflow/tensor2tensor/blob/master/docs/tutorials/asr_with_transformer.md](url)
when decoding. The training data is 100h+360h  and the model I used for decoding has the following status:

`INFO:tensorflow:Finished evaluation at 2018-05-25-13:32:33
INFO:tensorflow:Saving dict for global step 120002: global_step = 120002, loss = 1.0433111, metrics-librispeech/targets/accuracy = 0.633423, metrics-librispeech/targets/accuracy_per_sequence = 0.0, metrics-librispeech/targets/accuracy_top5 = 0.8924546, metrics-librispeech/targets/edit_distance = 0.5968527, metrics-librispeech/targets/neg_log_perplexity = -1.2044775
INFO:tensorflow:Stop training model as max steps reached
FLAGS.hparams_set:: transformer_librispeech
FLAGS.hparams:: batch_size=4096,`

I decoded the test data(those files are generated by t2t-datagen together with the training data), however the results are not quite as I expect
decode_from_file/61-70968-0001.flac.wav
Outputs: AND THAT THEY WERE TO BE SEEN THEY WERE ALL THE SAME AS THEY WERE TO BE ABLE TO SEE THEM
decode_from_file/61-70968-0002.flac.wav
Outputs: AND THAT THEY WERE ALL THE SAME AS THEY WERE
decode_from_file/61-70968-0003.flac.wav
Outputs: AND THAT THEY WERE ALL THE SAME
decode_from_file/61-70968-0004.flac.wav
Outputs: AND THAT THEY WERE ALL THE SAME AS THEY WERE ALL THE SAME

while the answers should be:
61-70968-0001 GIVE NOT SO EARNEST A MIND TO THESE MUMMERIES CHILD
61-70968-0002 A GOLDEN FORTUNE AND A HAPPY LIFE
61-70968-0003 HE WAS LIKE UNTO MY FATHER IN A WAY AND YET WAS NOT MY FATHER
61-70968-0004 ALSO THERE WAS A STRIPLING PAGE WHO TURNED INTO A MAID

### Question
1 is the model properly trained?  
2 what could be wrong if I want a decent result?

**Comment by chinnikrishna2231** [comment]
Hi,
I see similar output when I run decoder. I always get the same output in varying lenghts.
Is repo recently trained?

**Comment by vergilus** [comment]
the default network parameter is not sufficient for the task: I now use 100h+360h+500h as the training dataset(also larger batch) and larger/deeper network, now the result seems rather fine.
case closed.

**Comment by vergilus** [comment]
@chinnikrishna2231  try a larger/deeper training network(also training sets, the clean data is approximately 10W sentences). now it works for me. after 25W training steps on 7 GPU(batch size=4096), test on clean data and get ACC=90+

**Comment by chinnikrishna2231** [comment]
@vergilus 
These are the hyperparameters  I see by default. Can you share what you are using?
` hparams.num_heads = 4
  hparams.filter_size = 1024
  hparams.hidden_size = 256
  hparams.num_encoder_layers = 5
  hparams.num_decoder_layers = 3
  hparams.learning_rate = 0.15
  hparams.batch_size = 6000000`

**Comment by Qiaoxl** [comment]
@vergilus same question with chinnikrishna2231, can you give any suggestions on hparams?

**Comment by vergilus** [comment]
@Qiaoxl @chinnikrishna2231 
max_input_seq_length=3000    # 1550
num_heads=16    # 4
num_encoder_layers=6     # 5
num_decoder_layers=6     # 4
hidden_size=512    # 384
filter_size=4096     # 1024
relu_dropout=0.1     # 0.2
layers_prepostprocess_dropout=0.1    # 0.2
also try as large batch(for me: 8192 labels per batch) as you can.

**Comment by Qiaoxl** [comment]
@vergilus Thanks a lot.
